internal/resources/aws/db_instance.go

```diff
@@ -231,9 +235,70 @@ func (r *DBInstance) BuildResource() *schema.Resource {
 		})
 	}
 
+	if r.PerformanceInsightsEnabled {
+		if r.PerformanceInsightsLongTermRetention {
+			costComponents = append(costComponents, PerformanceInsightsLongTermRetentionCostComponent(r.Region, r.InstanceClass))
+		}
+
+		if r.MonthlyAdditionalPerformanceInsightsRequests == nil || *r.MonthlyAdditionalPerformanceInsightsRequests > 0 {
+			costComponents = append(costComponents,
+				PerformanceInsightsAPIRequestCostComponent(r.Region, r.MonthlyAdditionalPerformanceInsightsRequests))
+		}
```

Comment:
Just a thought, this already looks good. As alternative, we could move the conditions inside the cost component function and return an empty list. But this way it's more explicit, I guess.
```go
costComponents = append(costComponents, PerformanceInsightsAPIRequestCostComponent(r.Region, r.MonthlyAdditionalPerformanceInsightsRequests)...)
```
```go
func PerformanceInsightsAPIRequestCostComponent(region string, additionalRequests *int64) *schema.CostComponent {
	costComponents := []*schema.CostComponent{}
	if additionalRequests == nil || additionalRequests > 0 {
		return costComponents
	}
	// ...
}
```

Reply:
Yeah, I thought about that.  It seemed like it would be better to have the answer to the "When does this component actually get added?" question in resource itself 🤷‍♂️